tornado/wsgi.py

```diff
@@ -89,11 +93,33 @@ async def main():
     The `tornado.web.FallbackHandler` class is often useful for mixing
     Tornado and WSGI apps in the same server.  See
     https://github.com/bdarnell/django-tornado-demo for a complete example.
+
+    `WSGIContainer` supports executing the WSGI application in custom executors
+    using `IOLoop.run_in_executor`. The default executor uses
+    `tornado.concurrent.dummy_executor` which works synchronously, but other
+    executors subclassing `concurrent.futures.Executor` may be used. To execute
+    WSGI application code in separate threads in an event-loop compatible way
+    use::
+
+        async def main():
+            executor = concurrent.futures.ThreadPoolExecutor()
+            # ^-- Execute requests in separate threads.
+            container = tornado.wsgi.WSGIContainer(simple_app, executor)
+            # Initialize the WSGI container with custom executor --^
+            http_server = tornado.httpserver.HTTPServer(container)
+            http_server.listen(8888)
+            await asyncio.Event().wait()
+
+    Running the WSGI app with a `ThreadPoolExecutor` remains *less scalable*
```

Comment:
Why do you say that WSGIContainer with a ThreadPoolExecutor is less scalable than the alternatives? My expectation has been that we'd be comparable to them - not the best, but close enough that we could stop actively discouraging the use of WSGIContainer. (of course this assumes that you're running multiple processes as well)

Reply:
Tbh, I don't remember. I think I was just parroting what was written before. I'm not sure how gunicorn/uwsgi do multithreading but it wouldn't surprise me to learn that it's more efficient than ThreadPoolExecutor.